src/ui/content/login.tsx

```diff
@@ -1,44 +1,77 @@
-import React from 'react'
+import React, { useState } from 'react'
+import gql from 'graphql-tag'
+import { useMutation } from '@apollo/react-hooks'
 
-type MyProps = { }
-type MyState = { email: string, password: string}
-
-//validation?
+interface LoginInput {
+  password: string,
+  email: string
+}
 
-class LogForm extends React.Component<MyProps, MyState> {
-  constructor(props: any) {
-    super(props);
-    this.state = {email: '', password:''};
+type LoginResponse = {
+  token: string
+}
 
-    this.handleChange = this.handleChange.bind(this);
-    this.handleSubmit = this.handleSubmit.bind(this);
+const LOGIN_M = gql`
+  mutation Login($data: LoginInput!) {
+    loginUser(data: $data) {
+      token
+    }
   }
+`
+function LogForm() {
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
 
-  handleChange(event: any) {
-    if(event.target.name === "email")
-      this.setState({email: event.target.value})
-    else
-      this.setState({password: event.target.value})
+  const [login, { data:response, error }] = useMutation(LOGIN_M, { errorPolicy: 'all' })
+  
+  //why am I not getting the first click login data, only from the 2nd click
```

Comment:
1st submit: I am getting nothing back
2nd submit: I am getting back the data from the first submit

Reply:
The response is not updated until the next render, also the object is replaced then, but the response inside handleSubmit will keep referring to the old variable.

You run into this because you try to use this in an imperative way, while react and hooks are meant to be used in a declarative way.

In the options of the useMutation hook you can declare what should happen if the mutation was completed successful by providing a onCompleted callback.

Same is for the error handling, adding it to the html instead of showing an alert matches better with how react and hooks are working.